src/components/copy-link.ts

```diff
@@ -24,3 +24,31 @@ export function setupCopyLink({ element }: CopyLinkOptions) {
 		}
 	}
 }
+
+// Get the most appropriate URL to share.
+function getShareableUrl(): string {
+	if (window.parent === window) {
+		return window.location.href;
+	}
+
+	try {
+		return window.parent.location.href;
+	} catch {
+		// For cross-origin errors in iframe scenarios, fall back to referrer.
+		const referrer = document.referrer;
+
+		if (referrer) {
+			const referrerUrl = new URL(referrer);
+
+			referrerUrl.hash = window.location.hash;
+			referrerUrl.search = window.location.search;
+
+			// NOTE: This relies on the parent website's implementation, so we need to ensure it's consistent.
+			referrerUrl.pathname = '/demo/';
```

Comment:
[note] It'd be great if anyone could provide another way to avoid the hard-coded path.